common/subdevice-model.cpp

```diff
@@ -612,63 +626,287 @@ namespace rs2
-                    if (ImGui::Combo(label.c_str(), &ui.selected_shared_fps_id, fps_chars.data(),
-                        static_cast<int>(fps_chars.size())))
-                    {
-                        res = true;
-                    }
+                    ImGui::Combo(label.c_str(), &ui.selected_format_id[f.first], formats_chars.data(),
+                        static_cast<int>(formats_chars.size()));
                     ImGui::PopStyleColor();
                     ImGui::PopItemWidth();
                 }
-
                 ImGui::SetCursorPosX(col0);
+                // FPS
+                // Draw combo-box with all FPS options for this device
+                if (!show_single_fps_list && !fpses_per_stream[f.first].empty() && stream_enabled[f.first])
+                {
+                    auto fps_chars = get_string_pointers(fpses_per_stream[f.first]);
+                    ImGui::Text("Frame Rate (FPS):");
+                    streaming_tooltip();
+                    ImGui::SameLine(); ImGui::SetCursorPosX(col1);
+
+                    label = rsutils::string::from() << "##" << s->get_info(RS2_CAMERA_INFO_NAME)
+                        << s->get_info(RS2_CAMERA_INFO_NAME) << f.first << " fps";
+
+                    if (streaming)
+                    {
+                        ImGui::Text("%s", fps_chars[ui.selected_fps_id[f.first]]);
+                        streaming_tooltip();
+                    }
+                    else
+                    {
+                        ImGui::PushItemWidth(-1);
+                        ImGui::PushStyleColor(ImGuiCol_TextSelectedBg, { 1,1,1,1 });
+                        ImGui::Combo(label.c_str(), &ui.selected_fps_id[f.first], fps_chars.data(),
+                            static_cast<int>(fps_chars.size()));
+                        ImGui::PopStyleColor();
+                        ImGui::PopItemWidth();
+                    }
+                    ImGui::SetCursorPosX(col0);
+                }
+            }
+            else
+            {
+                //ImGui::NextColumn();
             }
         }
 
-        if (draw_streams_selector)
+        return res;
+    }
+
+    bool subdevice_model::draw_res_stream_formats(std::string& error_message, std::string& label, std::function<void()> streaming_tooltip, float col0, float col1)
+    {
+        bool res = false;
+
+        if (!ui.is_multiple_resolutions)
         {
-            if (!streaming)
+            return false;
+        }
+
+        int depth_res_id, ir1_res_id, ir2_res_id;
+        get_depth_ir_mismatch_resolutions_ids(depth_res_id, ir1_res_id, ir2_res_id);
+
+        std::vector<uint32_t> stream_types_ids;
+        stream_types_ids.push_back(depth_res_id);
+        stream_types_ids.push_back(ir1_res_id);
+        for (auto&& stream_type_id : stream_types_ids)
+        {
+            // resolution
+            // Draw combo-box with all resolution options for this stream type
```

Comment:
Do you think we can break it into 2 functions according to your comments?
* `Draw combo-box with all resolution options for this stream type`
* `Draw combo-box with all format options for current stream type`